public/view/main.jsx

```diff
@@ -39,33 +41,38 @@ const SUPPORTED_COUNTRIES = ['br', 'co', 'pa'];
 // ugliness here because AppMain hasn't yet been instantiated.
 var main_instance = null;
 var map_controller = null;
-var rerender = function() {
+var _base_rerender = function() {
   if (main_instance !== null) {
     main_instance.forceUpdate();
   }
 };
-var rerender_and_redraw = function() {
-  rerender();
+var rerender = _.debounce(_base_rerender, 16);  // 16 milliseconds is 60 frames per second ... fast enough
+var rerender_and_redraw = _.debounce(function() {
+  _base_rerender();
   if (map_controller !== null) {
     map_controller.redraw();
   }
-};
+}, 16);
 
 // NOTE: we could model resize state formally, but this'll do for now
 window.addEventListener('resize', rerender);
 
-var selection_ee = new SelectionEvents.SelectionEventEmitter();
-
-var loading_status = new LoadingStatusModel(rerender);
+var loading_status = new LoadingStatusModel();
 var api_client = new APIClient(loading_status);
-var epi_data_store = new EpiDataStore(rerender_and_redraw);
-var econ_data_store = new EconDataStore(rerender_and_redraw);
-var mobility_data_store = new MobilityDataStore(rerender_and_redraw, api_client);
-var weather_data_store = new WeatherDataStore(rerender_and_redraw, api_client, SUPPORTED_COUNTRIES);
-var selected_layers = new SelectedLayers(rerender_and_redraw);
-var selected_countries = new SelectedCountries(selection_ee, SUPPORTED_COUNTRIES);
-var selected_date = new SelectedDate(selection_ee, weather_data_store);
-var selected_admins = new SelectedAdmins(selection_ee);
+var epi_data_store = new EpiDataStore();
+var econ_data_store = new EconDataStore();
+var mobility_data_store = new MobilityDataStore(api_client);
+var weather_data_store = new WeatherDataStore(api_client, SUPPORTED_COUNTRIES);
+var selected_layers = new SelectedLayers();
+var selected_countries = new SelectedCountries(SUPPORTED_COUNTRIES);
+var selected_date = new SelectedDate(weather_data_store);
+var selected_admins = new SelectedAdmins();
+
+loading_status.emitter.any_event_listener(rerender);
+epi_data_store.emitter.any_event_listener(rerender_and_redraw);
+econ_data_store.emitter.any_event_listener(rerender_and_redraw);
+mobility_data_store.emitter.any_event_listener(rerender_and_redraw);
+weather_data_store.emitter.any_event_listener(rerender_and_redraw);
```

Comment:
Is the use of `any_event_listener` just a shortcut? It seems like all of this wiring could specify the specific type of event we expect?